### DISCUSSION TITLE
MOOSE Compiling Issue -Libmesh Configuration

### DISCUSSION THREAD

**Discussion by AhmedAlmetwally** [idaholab/moose#17993] [Q&A Getting Started] [answered]
I downloaded MOOSE on MAC and activated it through conda, but libmesh did not load and I receive the following:

```
almeag-mac@FN601235 ~ % cd ~/projects/moose/test
make -j 4
./run_tests -j 4
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Using HIT from /Users/almeag-mac/projects/moose/framework/contrib/hit
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Compiling C++ (in opt mode) /Users/almeag-mac/projects/moose/framework/build/unity_src/src_Unity.C...
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Compiling C++ (in opt mode) /Users/almeag-mac/projects/moose/framework/build/unity_src/linesearches_Unity.C...
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
make: *** [/Users/almeag-mac/projects/moose/framework/build/unity_src/src_Unity..opt.lo] Error 127
make: *** Waiting for unfinished jobs....
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
make: *** [/Users/almeag-mac/projects/moose/framework/build/unity_src/linesearches_Unity..opt.lo] Error 127
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Using HIT from /Users/almeag-mac/projects/moose/framework/contrib/hit
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Building and linking hit...
bash: -std=c++11: command not found
make: *** [hit] Error 127
failed to import pyhit - try running "make hit" in the $MOOSE_DIR/test directory.
```

Following the steps on: https://mooseframework.inl.gov/getting_started/installation/conda.html

**Comment by loganharbour** [comment]
Did you run

```
conda create --name moose moose-libmesh moose-tools
conda activate moose
```

as listed on the page you linked?

The fact that your install looks for libmesh within the moose directory and not a conda directory leads me to believe that you did not.

**Reply by AhmedAlmetwally** [reply]
I did

**Reply by loganharbour** [reply]
And what is the result of

```
echo $LIBMESH_DIR
```

?

**Reply by AhmedAlmetwally** [reply]
I removed MOOSE and tried to reinstall it several times and I got repetitively the following error:
CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://mooseframework.org/conda/moose/osx-64/current_repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
'https://mooseframework.org/conda/moose/osx-64'

**Reply by loganharbour** [reply]
Please see this announcement and follow the instructions: https://groups.google.com/g/moose-announce/c/6-eXRa3mJmA/m/lGbpo31RAQAJ

After following those commands, you should be able to remove any old environments with the name moose:
```
conda env remove --name moose
```

and then re-add it:
```
conda create --name moose moose-libmesh moose-tools
```

Go to compile, and before doing so activate the environment:
```
conda activate moose
cd ~/projects/moose/test
make -j 4
```

**Reply by AhmedAlmetwally** [reply]
I did and received the same error. I ran 
almeag-mac@FN601235 test % echo $LIBMESH_DIR
almeag-mac@FN601235 test %

**Reply by AhmedAlmetwally** [reply]
It return anything.

**Reply by AhmedAlmetwally** [reply]
make: *** Waiting for unfinished jobs....
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
make: *** [/Users/almeag-mac/projects/moose/framework/src/utils/RankFourTensor..opt.lo] Error 127
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/libtool: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Using HIT from /Users/almeag-mac/projects/moose/framework/contrib/hit
/bin/sh: /Users/almeag-mac/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
Building and linking hit...
bash: -std=c++11: command not found
make: *** [hit] Error 127
failed to import pyhit - try running "make hit" in the $MOOSE_DIR/test directory.

**Reply by loganharbour** [reply]
You're still not properly activating the conda environment. If the conda environment is active, `$LIBMESH_DIR` will be set to a directory.

I suggest that you remove miniconda with `rm -rf ~/miniconda3` and follow the directions again. You're missing a step somewhere.

**Reply by AhmedAlmetwally** [reply]
I use this command:
curl -L -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
bash Miniconda3-latest-MacOSX-x86_64.sh -b -p ~/miniconda3

**Reply by loganharbour** [reply]
The remainder of the instructions are very important.

If LIBMESH_DIR is not set, you are most likely not following them in their entirety.

**Comment by GiudGiud** [comment, marked as answer]
Assuming the conda reinstall went fine (to check could you please run `conda list moose`, thanks!)

I think you're just missing the `conda activate moose` step

**Reply by AhmedAlmetwally** [reply]
almeag-mac@FN601235 test % conda activate moose

CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.

**Reply by loganharbour** [reply]
Please follow the instructions - this is explained clearly in the end of the section "Install MOOSE Conda Packages" and there are even frequently asked questions about this issue on the page.

**Reply by GiudGiud** [reply]
@AhmedAlmetwally 
did you do what the error message says?
```conda init zsh```
then close and reopen the shell. 

Then `conda activate moose` should work.
Please let us know how this goes

**Reply by AhmedAlmetwally** [reply]
@GiudGiud @loganharbour, Thank you so much! It compiled now properly conda initiation.
